### DISCUSSION TITLE
Why is Node v22 needed?

### DISCUSSION THREAD

**Discussion by ExpandOcean** [zen-fs/core#198] [Q&A] [answered]
Hello, the engine requires Node.js version 22 or higher. Is there a specific feature that needs this? The current default LTS version of Node.js is 20.

**Comment by james-pre** [comment, marked as answer]
@ExpandOcean,

Thanks for the question.

As mentioned in the [0.14.0 release notes](https://github.com/zen-fs/core/releases/tag/v0.14.0), the change to need Node v22 is for [`Promise.withResolvers`](https://mdn.io/Promise.withResolvers). If you polyfill `Promise.withResolvers` you should be able to use a lower Node version. 

I think it may be a good idea to add the polyfill in ZenFS since it is trivial and would save this headache. Especially since the engine field of the NPM package wouldn't need to be 22 then.

Just to show how easy the polyfill is:
```ts
Promise.withResolvers ??= function<T>(): PromiseWithResolvers<T> {
    let resolve, reject;
    const promise = new Promise((_resolve, _reject) => {
        resolve = _resolve;
        reject = _reject;
    });
    return { promise, resolve, reject };
}
```

**Comment by ExpandOcean** [comment]
Thanks for your reply and the new release! I like this library and want to use it in an existing project but there will be a conflict if Node>20, the latest release should solve my problem. Thanks!
